zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateCompensationSubscriptionTest.java

```diff
@@ -0,0 +1,520 @@
+                    .startEvent()
+                    .serviceTask("A", t -> t.zeebeJobType("A"))
+                    .boundaryEvent(
+                        "boundary1",
+                        b ->
+                            b.compensation(
+                                c -> c.serviceTask("undoA", t -> t.zeebeJobType("undoA"))))
+                    .moveToActivity("A")
+                    .serviceTask("B", t -> t.zeebeJobType("B"))
+                    .intermediateThrowEvent(
+                        "boundary_throw", ic -> ic.compensateEventDefinition().activityRef("A"))
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .serviceTask("C", t -> t.zeebeJobType("C"))
+                    .endEvent()
+                    .done())
+            .deploy();
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+    final long sourceProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, processId);
+
+    final var processInstanceKey = engine.processInstance().ofBpmnProcessId(processId).create();
+
+    engine.job().ofInstance(processInstanceKey).withType("A").complete();
+
+    RecordingExporter.compensationSubscriptionRecords()
+        .withProcessInstanceKey(processInstanceKey)
+        .withIntent(CompensationSubscriptionIntent.CREATED)
+        .await();
+
+    // when
+    engine
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("B", "C")
+        .migrate();
+
+    // then
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.PROCESS)
+                .getFirst())
+        .describedAs("Expect that the process instance is migrated")
+        .isNotNull();
+
+    Assertions.assertThat(
+            RecordingExporter.compensationSubscriptionRecords()
+                .withIntent(CompensationSubscriptionIntent.DELETED)
+                .withProcessInstanceKey(processInstanceKey)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that the process definition is updated")
+        .hasProcessDefinitionKey(sourceProcessDefinitionKey)
+        .describedAs("Expect that the compensable activity is updated")
+        .hasCompensableActivityId("A")
+        .describedAs("Expect that the compensation handler id is unchanged")
+        .hasCompensationHandlerId("undoA")
+        .describedAs("Expect that the compensation boundary event in the source is deleted")
+        .isNotNull();
```

Comment:
🔧 I'll put this at the beginning of the chain

Reply:
In other migration tests as well, we first make sure the overall migration was successful. However, I think, we can completely remove the assertion since the `migrate()` already awaits MIGRATED event to be written. Updated.